streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -479,9 +477,26 @@ void commit(final boolean startNewTransaction) {
             throw new TaskMigratedException(this, error);
         }
 
+        markCommitDone(time.nanoseconds() - startNs);
+    }
+
+    @Override
+    public Map<TopicPartition, OffsetAndMetadata> getPendingOffsets() {
+        final Map<TopicPartition, OffsetAndMetadata> consumedOffsetsAndMetadata = new HashMap<>(consumedOffsets.size());
+        for (final Map.Entry<TopicPartition, Long> entry : consumedOffsets.entrySet()) {
+            final TopicPartition partition = entry.getKey();
+            final long offset = entry.getValue() + 1;
+            consumedOffsetsAndMetadata.put(partition, new OffsetAndMetadata(offset));
+            stateMgr.putOffsetLimit(partition, offset);
+        }
+        return consumedOffsetsAndMetadata;
+    }
+
+    @Override
+    public void markCommitDone(long commitLatency) {
```

Comment:
Do these two functions have to be on AbstractTask as well?

Reply:
Since we need to access this function in `AssignedTasks`, either we generify the two functions, or we need to cast the task on to StreamTask in `AssignedTasks` instead.